client-app/shared/catalog/components/product-price-block.vue

```diff
@@ -3,122 +3,114 @@
-          </span>
-        </button>
-
-        <div class="flex w-3/5">
-          <div class="w-1/3">
-            <VcPopover y-offset="20" trigger="click" @toggle="handleShareProductPopoverToggle">
-              <template #trigger>
-                <div
-                  class="flex h-14 cursor-pointer select-none items-center justify-center space-x-2 border-r p-2 hover:bg-gray-100"
-                >
-                  <VcIcon
-                    name="share"
-                    :size="18"
-                    :class="{
-                      'text-[--color-primary-500]': !shareProductPopoverShown,
-                      'text-[--color-neutral-400]': shareProductPopoverShown,
-                    }"
-                  />
-                </div>
-              </template>
-
-              <template #content="{ close }">
-                <div
-                  class="rounded border bg-white p-5 shadow-lg before:absolute before:-top-2 before:left-[calc(50%-0.5rem)] before:h-4 before:w-4 before:rotate-45 before:border-l before:border-t before:bg-inherit"
-                >
-                  <h3 class="flex justify-between text-lg font-bold">
-                    <span class="flex grow">
-                      {{ $t("shared.catalog.product_details.share_product_label") }}
-                    </span>
-
-                    <button
-                      class="-me-1 flex p-1 text-[--color-danger-400] hover:text-[--color-danger-700]"
-                      type="button"
-                      @click="close()"
-                    >
-                      <VcIcon name="x" size="sm" />
-                    </button>
-                  </h3>
-
-                  <div class="mt-5 flex items-center space-x-6">
-                    <a
-                      v-for="socialSharingService in $cfg.social_sharing_services"
-                      :key="socialSharingService.name"
-                      target="_blank"
-                      :href="getProductSocialShareUrl(socialSharingService.url_template, pageUrl)"
-                    >
-                      <img
-                        class="rounded-sm"
-                        width="40"
-                        height="40"
-                        :src="socialSharingService.icon"
-                        :alt="socialSharingService.name"
-                      />
-                    </a>
-                  </div>
-                </div>
-              </template>
-            </VcPopover>
-          </div>
-
-          <div class="w-1/3">
-            <a
-              :href="mailToLink"
-              target="_blank"
-              class="flex h-14 cursor-pointer select-none items-center justify-center border-r p-2 hover:bg-gray-100"
+      <div class="flex select-none divide-x print:hidden">
+        <AddToList class="flex w-1/5 flex-col justify-stretch" :product="product" tooltip-placement="bottom">
+          <template #default="{ openModal, isInWishlist, isAuthenticated }">
```

Comment:
1. I would suggest to use `isAuthenticated` directly from `useUser` composable.
2. We also have `product` here so we don't need to pass product to modal and then get `product.isInWishlist` back, instead just directly use `product.isInWishlist`.
3. The last one is to open modal. I think we can leave it as is or move to product composable, it's up to you.

Reply:
https://virtocommerce.atlassian.net/browse/ST-5654